diff --git a/src/bin/lwan/main.c b/src/bin/lwan/main.c
index 6f6ba92..b30f64e 100644
--- a/src/bin/lwan/main.c
+++ b/src/bin/lwan/main.c
@@ -190,7 +190,6 @@ main(int argc, char *argv[])
         if (sj.chroot_path) {
             root = "/";
         }
-        lwan_straitjacket_enforce(&sj);
 
         lwan_init_with_config(&l, &c);
 
@@ -201,7 +200,6 @@ main(int argc, char *argv[])
         lwan_set_url_map(&l, map);
         break;
     case ARGS_USE_CONFIG:
-        lwan_straitjacket_enforce(&sj);
         if (c.config_file_path)
             lwan_init_with_config(&l, &c);
         else
@@ -212,6 +210,7 @@ main(int argc, char *argv[])
         goto out;
     }
 
+    lwan_straitjacket_enforce(&sj);
     lwan_main_loop(&l);
     lwan_shutdown(&l);
 
